packages/camel-catalog/catalog-generator/src/main/java/io/kaoto/camelcatalog/CamelCatalogVersionLoader.java

```diff
@@ -0,0 +1,235 @@
+package io.kaoto.camelcatalog;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.JarURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+import org.apache.camel.catalog.CamelCatalog;
+import org.apache.camel.catalog.DefaultCamelCatalog;
+import org.apache.camel.catalog.VersionManager;
+import org.apache.camel.catalog.maven.MavenVersionManager;
+
+public class CamelCatalogVersionLoader {
+    private final VersionManager VERSION_MANAGER = new MavenVersionManager();
+    private CamelCatalog camelCatalog = new DefaultCamelCatalog(false);
+    private String camelYamlDSLSchema;
+    private String kubernetesSchema;
+    private List<String> kameletBoundaries = new ArrayList<>();
+    private List<String> kamelets = new ArrayList<>();
+    private List<String> camelKCRDs = new ArrayList<>();
+    private Map<String, String> localSchemas = new HashMap<>();
+
+    public CamelCatalog getCamelCatalog() {
+        return camelCatalog;
+    }
+
+    public String getCamelYamlDslSchema() {
+        return camelYamlDSLSchema;
+    }
+
+    public List<String> getKameletBoundaries() {
+        return kameletBoundaries;
+    }
+
+    public List<String> getKamelets() {
+        return kamelets;
+    }
+
+    public String getKubernetesSchema() {
+        return kubernetesSchema;
+    }
+
+    public List<String> getCamelKCRDs() {
+        return camelKCRDs;
+    }
+
+    public Map<String, String> getLocalSchemas() {
+        return localSchemas;
+    }
+
+    public boolean loadCamelCatalog(String version) {
+        camelCatalog.setVersionManager(VERSION_MANAGER);
+
+        return camelCatalog.loadVersion(version);
+    }
+
+    public boolean loadCamelYamlDsl(String version) {
+        boolean isCamelYamlDslLoaded = loadDepedencyInClasspath(Constants.APACHE_CAMEL_ORG,
+                Constants.CAMEL_YAML_DSL_PACKAGE,
+                version);
+
+        ClassLoader classLoader = VERSION_MANAGER.getClassLoader();
+        URL resourceURL = classLoader.getResource(Constants.CAMEL_YAML_DSL_ARTIFACT);
+        if (resourceURL == null) {
+            return false;
+        }
```

Comment:
I think that in case the resource is not found we want to have an error to detect for instance that the filename has changed.

Reply:
This is still pending until we decide how to use this code